bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/fileformat/FileFormatResource.java

```diff
@@ -485,4 +622,179 @@ private List<Thing> getThingsOrDiscoveryResult(List<String> thingUIDs) {
             return simulateThing(discoveryResult, thingType);
         }).toList();
     }
+
+    private boolean convertFromFileFormatDTO(FileFormatDTO data, List<Thing> things, List<String> errors) {
+        boolean ok = true;
+        if (data.things != null) {
+            for (ThingDTO thingBean : data.things) {
+                ThingUID thingUID = thingBean.UID == null ? null : new ThingUID(thingBean.UID);
+                ThingTypeUID thingTypeUID = new ThingTypeUID(thingBean.thingTypeUID);
+
+                ThingUID bridgeUID = null;
+
+                if (thingBean.bridgeUID != null) {
+                    bridgeUID = new ThingUID(thingBean.bridgeUID);
+                    if (thingUID != null && (!thingUID.getBindingId().equals(bridgeUID.getBindingId())
+                            || !thingUID.getBridgeIds().contains(bridgeUID.getId()))) {
+                        errors.add("Thing UID '" + thingUID + "' does not match bridge UID '" + bridgeUID + "'");
+                        ok = false;
+                        continue;
+                    }
```

Comment:
```suggestion
```
Please remove this check because a thing uid like `mqtt:topic:mylight` which doesn't include the bridge uid is allowed. All my mqtt things are defined this way (my preference).

By allowing it, instead of returning an error, it will also give the user a chance to make changes to fix the "issue". In any case, I don't think it's the role of the converter to perform this validation at this point in the code.

Reply:
This part of code was more or less duplicated from elsewhere, probably from the API to create a thing. Let me check.

Reply:
It was part of DiscoveryResultBuilder and ThingResource, which creates a managed Thing from UI.

File based Things both DSL and JRuby do not place such restrictions.

If we keep it here, then we won't be able to create YAML for DSL/Ruby created Things that don't satisfy this criteria, despite the Thing working just fine otherwise.

Removing it here won't be a problem for managed things, because Thing UID cannot be changed for existing Thing, and in the case of Thing creation, it will be rejected when user tries to submit the Thing.

Reply:
Constraint removed. I just kept the check about the same binding.

Reply:
Thank you. It works.